### PR TITLE
Document the new 24H2 APIs

### PR DESCRIPTION
**Pull request by lhecker**
Adds documentation for:
* Console Allocation Policy
* `AllocConsoleWithOptions` and its accompanying struct
* `ReleasePseudoConsole`
* Changed behavior of `ClosePseudoConsole`

### CODE REVIEW

File: docs/TOC.md [hunk 1]
```diff
@@ -1,140 +1,143 @@
...
+### [CONSOLE_SCREEN_BUFFER_INFO structure](console-screen-buffer-info-str.md)
+### [CONSOLE_SCREEN_BUFFER_INFOEX structure](console-screen-buffer-infoex.md)
+### [CONSOLE_SELECTION_INFO structure](console-selection-info-str.md)
+### [COORD structure](coord-str.md)
+### [FOCUS_EVENT_RECORD structure](focus-event-record-str.md)
+### [INPUT_RECORD structure](input-record-str.md)
+### [KEY_EVENT_RECORD structure](key-event-record-str.md)
+### [MENU_EVENT_RECORD structure](menu-event-record-str.md)
+### [MOUSE_EVENT_RECORD structure](mouse-event-record-str.md)
+### [SMALL_RECT structure](small-rect-str.md)
```
Comment (lhecker) [inline docs/TOC.md:140]: I sorted these alphabetically.

File: docs/releasepseudoconsole.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+**ReleasePseudoConsole** solves this problem: It allows you to relinquish your half of the ownership. This in turn allows the pseudoconsole to automatically exit as soon as all clients have disconnected. All you need to do now is to read from or write to your output and input pipe handles until they return a failure. This indicates to you that all clients have disconnected and the the pseudoconsole is about to exit on its own.
+
+> [!WARNING]
+> **ReleasePseudoConsole** does not deallocate the memory associated with the **HPCON**. You must still call **ClosePseudoConsole** once you're done using the **HPCON** instance. See [ClosePseudoConsole](closepseudoconsole.md) for important information about its correct usage.
+
+## Examples
+
+For a full walkthrough on using this function to establish a pseudoconsole session, please see [Creating a Pseudoconsole Session](creating-a-pseudoconsole-session.md).
```
Comment (DHowett) [inline docs/releasepseudoconsole.md:58]: This PR doesn't change this file - should it?

File: docs/releasepseudoconsole.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+## Return value
+
+Type: **HRESULT**
+
+If this method succeeds, it returns **S_OK**. Otherwise, it returns an **HRESULT** error code.
+The call is not expected to fail unless the **hPC** argument is invalid in which case it returns **E_INVALIDARG**.
+
+## Remarks
+
+Before this method was introduced, the initial pseudoconsole API was inherently flawed: **ClosePseudoConsole** would only exit until after the pseudoconsole has finished using the output pipe. This meant that the pseudoconsole effectively kept your application alive. The **HPCON** handle owned by your application on the other hand kept the pseudoconsole session alive until **ClosePseudoConsole** was called. This created a lifetime and ownership loop. The way applications were expected to break this loop up was by manually waiting for the spawned terminal processes to exit and then call **ClosePseudoConsole**. This made the API failure prone and resulted in shutdown deadlocks in various applications.
```
Comment (DHowett) [inline docs/releasepseudoconsole.md:49]: I'd rephrase the "inherently flawed" part... but i am not sure how

Comment (DHowett) [reply docs/releasepseudoconsole.md:49]: (We don't usually editorialize in the docs!)

**Comment by floh96** [comment]
@lhecker can you make sure this gets merged? Thanks

File: docs/console-allocation-policy.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+```
+
+The **IMAGE_SUBSYSTEM_WINDOWS_CUI** type informs shells that they need to block until your application has finished executing, while the application manifest informs the operating system to skip allocating a console.
+
+## Requirements
+
+| &nbsp; | &nbsp; |
+|-|-|
+| Minimum supported client | Windows 11 24H2 (build 26100) \[desktop apps only\] |
+| Minimum supported server | n/a |
```
Comment (DHowett) [inline docs/console-allocation-policy.md:43]:
```suggestion
| Minimum supported server | Windows Server 2025 (build 26100) |
```

File: docs/allocconsolewithoptions.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+The *result* parameter will be set to **ALLOC\_CONSOLE\_RESULT\_EXISTING\_CONSOLE** in that case.
+
+A process can use the [**FreeConsole**](freeconsole.md) function to detach itself from its current console. A console is closed when the last process attached to it terminates or calls **FreeConsole**.
+
+## Requirements
+
+| &nbsp; | &nbsp; |
+|-|-|
+| Minimum supported client | Windows 11 24H2 (build 26100) \[desktop apps only\] |
+| Minimum supported server | n/a |
```
Comment (DHowett) [inline docs/allocconsolewithoptions.md:65]: same note about server 2025 applies to all of these

**Review by DHowett — APPROVED** [review]
